lms/static/scripts/frontend_apps/components/StudentSelector.js

```diff
@@ -0,0 +1,97 @@
+import { createElement } from 'preact';
+import propTypes from 'prop-types';
+
+/**
+ * A student navigation tool which shows a student selection list and a previous and next button.
+ */
+
+export default function StudentSelector({
+  onSelectStudent,
+  selectedStudentIndex,
+  students,
+}) {
+  // Disable the next button if at the end of the list.
+  const hasNextStudent = selectedStudentIndex + 1 < students.length;
+  // Disable the previous button if at the start of the list.
+  const hasPrevStudent = selectedStudentIndex > 0;
+
+  /**
+   * Select the next student in the list.
+   */
+  const onNextStudent = () => {
+    if (selectedStudentIndex + 1 < students.length) {
+      onSelectStudent(selectedStudentIndex + 1);
+    }
+  };
+  /**
+   * Select the previous student in the list.
+   */
+  const onPrevStudent = () => {
+    if (selectedStudentIndex > 0) {
+      onSelectStudent(selectedStudentIndex - 1);
+    }
+  };
+
+  /**
+   * Build the <select> list from the current array of students.
+   */
+  const studentList = () => {
+    const options = students.map((student, i) => (
+      <option
+        key={`student-${i}`}
+        selected={selectedStudentIndex === i}
+        value={i}
+      >
+        {student.displayName}
+      </option>
+    ));
+
+    if (selectedStudentIndex < 0) {
+      // add the default option (no selection)
+      options.push(
+        <option hidden disabled selected>
```

Comment:
The way these attrs work together make it impossible to get out of user-focused mode once you're in it, unfortunately. It also makes it confusing when first opening the `select` because it looks like the first student is selected:

![image](https://user-images.githubusercontent.com/439947/65980578-76e36080-e445-11e9-8ed5-96fd9b5332b6.png)

Every BlackBoard instructor launching an assignment is going to get chucked into this mode, so I think we need to have an escape hatch back to viewing everyone, perhaps a default option like "All Students"?

Reply:
Perfect